src/edu/ucsb/nceas/metacat/dataone/CNodeService.java

```diff
@@ -87,2025 +55,2135 @@
+            D1AuthHelper authDel = new D1AuthHelper(request, pid, "4861", "????");
+            authDel.doCNOnlyAuthorization(session);
+        }
+        // the identifier can't be an SID
+        try {
+            if (IdentifierManager.getInstance().systemMetadataSIDExists(pid)) {
+                throw new InvalidRequest(
+                    "4863", "The provided identifier " + pid.getValue()
+                    + " is a series id which is not allowed.");
+            }
+        } catch (SQLException sqle) {
+            throw new ServiceFailure(
+                "4862", "Couldn't determine if the pid " + pid.getValue() + " is a series id since "
+                + sqle.getMessage());
+        }
+
+        // verify that guid == SystemMetadata.getIdentifier()
+        logMetacat.debug(
+            "Comparing guid|sysmeta_guid: " + pid.getValue() + "|" + sysmeta.getIdentifier()
+                .getValue());
+        if (!pid.getValue().equals(sysmeta.getIdentifier().getValue())) {
+            throw new InvalidRequest("4863", "The identifier in method call (" + pid.getValue()
+                + ") does not match identifier in system metadata (" + sysmeta.getIdentifier()
+                .getValue() + ").");
+        }
+
+        //check if the sid is legitimate in the system metadata
+        //checkSidInModifyingSystemMetadata(sysmeta, "4864", "4862");
+        Identifier sid = sysmeta.getSeriesId();
+        if (sid != null) {
+            if (!isValidIdentifier(sid)) {
+                throw new InvalidRequest(
+                    "4863", "The series id in the system metadata is invalid in the request.");
+            }
+        }
+
+        try {
+            logMetacat.debug("Checking if identifier exists...");
+            // Check that the identifier does not already exist
+            try {
+                if (IdentifierManager.getInstance().systemMetadataPIDExists(pid)) {
+                    throw new InvalidRequest("4863",
+                                             "The identifier is already in use by an existing "
+                                                 + "object.");
+                }
+            } catch (SQLException ee) {
+                throw new ServiceFailure(
+                    "4862", "Error inserting system metadata: " + ee.getClass() + ": "
+                    + ee.getMessage());
+            }
+            // insert the system metadata into the object store
+            logMetacat.debug("Starting to insert SystemMetadata...");
+            try {
+                //for the object whose authoriative mn is v1. we need reset the modification date.
+                //d1-sync already set the serial version. so we don't need do again.
+                D1NodeVersionChecker checker =
+                    new D1NodeVersionChecker(sysmeta.getAuthoritativeMemberNode());
+                String version = checker.getVersion("MNRead");
+                boolean changeModificationDate =
```

Comment:
Did you do the change or the editor does?

Reply:
disregard - will revert changes and recommit without code cleanup